src/FuelClient.cc

```diff
@@ -657,9 +663,17 @@ Result FuelClient::DownloadModel(const ModelIdentifier &_id,
   }
   newId.SetVersion(version);
 
+  std::string zipData;
+  this->dataPtr->ZipFromResponse(resp, zipData);
+
   // Save
   // Note that the save function doesn't return the path
-  if (!this->dataPtr->cache->SaveModel(newId, resp.data, true))
+  if (!zipData.empty())
+  {
+    if (!this->dataPtr->cache->SaveModel(newId, zipData, true))
+      return Result(ResultType::FETCH_ERROR);
+  }
+  else
     return Result(ResultType::FETCH_ERROR);
```

Comment:
This can be simplified.

```suggestion
  if (zipData.empty() || !this->dataPtr->cache->SaveModel(newId, zipData, true))
      return Result(ResultType::FETCH_ERROR);
  }
```

Reply:
Fixed in [5e48ee4](https://github.com/gazebosim/gz-fuel-tools/pull/333/commits/5e48ee42b2830bfb87fc92e439b9e9a8329f5e77)